### ISSUE TITLE
Fix false positives in "Prefer finals over getters" rule.

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#35]
As @zoechi pointed out on #23, we need to refine our check to ensure we don't flag cases where the field is written to *outside* a constructor initializer.

In other words, this should produce a lint:
```
class A {
  var _a;
  var get a => _a; //LINT
  A(this._a);
}
```

But this should not:

```
class A {
  var _a;
  var get a => _a; //OK!
  A() {
    _init();
  }
  
  _init() {
   _a = 42;
  }
}
```

**Comment by xxgreg** [comment]
Given library based privacy, I assume you'll also need to check that _a is not set from anywhere in the library, not just from within the class as in the example above.

    library foo;
    part 'other.dart';
    class A {
       var _a;
    }

other.dart:    

    part of foo
    foo(bar) {
      bar._a = 'oi!';
    }

Also will there be a way to disable the lint if mirrors are used? (Another tricky case to consider is if generalised tear offs land)

**Comment by pq** [comment]
@xxgreg :+1: 

I started implementing this for real today and got far enough along to decide to disable it in the meantime.

Thanks for chiming in!

**Comment by pq** [comment]
Re-opened #23.  Tracking progress there.
